src/Currency.php

```diff
@@ -1315,25 +1558,47 @@ public static function fromNumericCode(int $code): self
         return new self($map[$code]);
     }
 
-    public static function addCurrency(string $code, string $displayName, int $numericCode, int $defaultFractionDigits, int $subUnit): void
-    {
+    public static function addCurrency(
+        string $code,
+        string $displayName,
+        int $numericCode,
+        int $defaultFractionDigits,
+        int $subUnit,
+        bool $deprecated = false
+    ): void {
         self::$currencies[$code] = [
             'display_name' => $displayName,
             'numeric_code' => $numericCode,
             'default_fraction_digits' => $defaultFractionDigits,
             'sub_unit' => $subUnit,
+            'deprecated' => $deprecated,
         ];
     }
 
+    /**
+     * Returns only active currencies.
+     */
     public static function getCurrencies(): array
+    {
+        return array_filter(
+            self::$currencies,
+            function ($currencyCode) {
```

Comment:
```suggestion
            static function (array $currency): bool {
```

Reply:
why static?

Reply:
isn't it static?

Reply:
micro-optimization 😄

Reply:
meaningful code :)

Reply:
i think it doesn't make much difference in this case